### PR TITLE
add auto tagging

### PR DESCRIPTION
**Pull request by jptosso**
This action should create a tag with release notes for each push to master

The idea is to track all fixes so we can easily track them in coraza

### CODE REVIEW

**Comment by jptosso** [comment]
Tested here: https://github.com/jptosso/test-actions

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+name: Create Release
+on:
+  push:
+    branches:
```
Comment (anuraaga) [inline .github/workflows/release.yml:4]: We should probably filter out non-logic files

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      - main
+      - master
+jobs:
+  release:
+    runs-on: ubuntu-20.04
```
Comment (anuraaga) [inline .github/workflows/release.yml:9]:
```suggestion
    runs-on: ubuntu-22.04
```

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
```
Comment (anuraaga) [inline .github/workflows/release.yml:12]:
```suggestion
        uses: actions/checkout@v3
```

**Comment by jcchavezs** [comment]
I am curious about this change. Do we want to do this? How will this
automation know if it is a major/minor/patch? Do we want to publish every
merge to master?

**Comment by jptosso** [comment]
I think we won't release minor or major versions for this library, there are only two functions, DetectXSS and DetectSQLi. So we should only deliver patches

File: .github/workflows/release.yml (hunk 1 above)
Comment (jptosso) [reply .github/workflows/release.yml:4]: Added:
```yaml
    paths-ignore:
      - "**/*.md"
      - "**/*.txt"
      - "**/*_test.go"
      - "LICENSE"
 ```

**Pull request merged**